addons/base_geolocalize/models/base_geocoder.py

```diff
@@ -87,8 +87,13 @@ def _call_openstreetmap(self, addr, **kw):
             return None
         url = 'https://nominatim.openstreetmap.org/search'
         try:
-            result = requests.get(url, {'format': 'json', 'q': addr}).json()
+            headers = {'User-Agent': 'Odoo (http://www.odoo.com/contactus)'}
+            response = requests.get(url, headers=headers, params={'format': 'json', 'q': addr})
             _logger.info('openstreetmap nominatim service called')
+            if response.status_code != 200:
+                _logger.error('Request to openstreetmap failed.\nCode: %s\nContent: %s' % (response.status_code, response.content))
+                raise Exception('openstreetmap status code: %s' % (response.status_code))
```

Comment:
Not sure we want to raise here. A regular end user won't be able to do anything about it, so log seems enough.